charts/kong/README.md

```diff
@@ -288,6 +293,124 @@ scenario and several settings that are useful when using multiple controllers:
 Standalone controllers require a database-backed Kong instance, as DB-less mode
 requires that a single controller generate a complete Kong configuration.
 
+### Hybrid mode
+
+Kong supports [hybrid mode
+deployments](https://docs.konghq.com/2.0.x/hybrid-mode/) as of Kong 2.0.0 and
+Kong Enterprise 2.1.0. These deployments split Kong nodes into control plane
+(CP) nodes, which provide the admin API and interact with the database, and
+data plane (DP) nodes, which provide the proxy and receive configuration from
+control plane nodes.
+
+The chart does not support deploying control and data plane nodes in a single
+release. Users should [create separate releases for each node
+type](#separate-admin-and-proxy-nodes), i.e. use separate control and data
+plane values.yamls that are installed separately.
+
+* If `.Values.env.role="control_plane"`, the release will create a Service
+  named `RELEASE-NAME-kong-cluster`.
```

Comment:
I'm unable to find a causal link between setting `.Values.env.role` to `control_plane` and the release creating such a service. How does it happen? Why is it important for the user to be mentioned here?

Where is the `.Values.env.role` field used in the chart? I'm unable to find any such place 🤔

I suggest defining `.Values.env.role` in the chart-level `values.yaml` and documenting it there.

Reply:
It used to, and I then removed it in favor of making the service and role stuff separate. That entails additional steps, but doesn't have any hidden functionality inconsistent with the other services' configuration, which I think is a reasonable tradeoff.

The original strategy made a bit more sense when I initially started work on this (a long long time ago--there was a significant pause due to priority shifts elsewhere), and the new strategy makes more sense in light of changes since to service handling.

Everything under `.Values.env` is translated into a `KONG_` environment variable, corresponding to [kong.conf](https://github.com/Kong/kong/blob/master/kong.conf.default) settings, so those generally aren't documented separately on their own (we do mention them in the context of specific configurations that need them, like here). There's a [rather large template helper](https://github.com/Kong/charts/blob/kong-1.7.0/charts/kong/templates/_helpers.tpl#L393-L581) that combines those values with others we set automatically from other sections of values.yaml. Those are then rendered in the container `env` sections of the deployment and migrations jobs, e.g. [for the main deployment](https://github.com/Kong/charts/blob/kong-1.7.0/charts/kong/templates/deployment.yaml#L75).

Reply:
Thanks for the explanation. Created PR #163 to extend the comment for `.Values.env` with a hint about the `KONG_` translation.